### DISCUSSION TITLE
[Deliver] Build number: $X does not exist

### DISCUSSION THREAD

**Discussion by EmDee** [fastlane/fastlane#17971] [General]
Has anyone else ran into this problem lately? Our CI fails during the deliver action stating that `Build number: $X does not exist` after running the pre-check successfully. However, when I check in `App Store Connect` the build number that `fastlane` apparently couldn't find is there.

Anyone else seeing this?

**Comment by mathaeus** [comment]
@EmDee yes, I'm also seeing this 😕 

Funnily enough I only get this locally though, on CI it's another problem we're running into, namely https://github.com/fastlane/fastlane/issues/17886

**Reply by EmDee** [reply]
Interesting. We're not using the Connect API key yet, but I guess I will wait until #17886 is fixed before doing so. Hopefully it's fixed before Apple enforces 2FA... 

I will be building a new release beginning next week so will see, if this was only a temporary problem for me.

**Reply by mathaeus** [reply]
I'm not 100% sure if any of this is really related to the API key or not.

I can only share my experience that yesterday and today trying to submit with `deliver` and API key did not work, while as a submission with `deliver` but without API key in between worked as expected.

**Reply by EmDee** [reply]
We‘re still getting this error when trying to submit.

**Reply by EmDee** [reply]
We are still using `fastlane 2.160.0`. Lets see if `2.172.0` did anything otherwise I will just open a bug report here. That seems to get more traction.

**Reply by mathaeus** [reply]
I also had this error (locally only) with `2.172.0` - see https://github.com/fastlane/fastlane/issues/17886#issuecomment-765252199

A bug report would be a good idea for sure 👍

**Reply by EmDee** [reply]
Yep, didn't change anything for us either.

But interesting though. Which CI are you guys using? Because our CI is basically our own local runner, so I guess both our issues are the same.

I'll create a bug report.

**Reply by mathaeus** [reply]
circleCI

Thanks for the bug report!

**Comment by EmDee** [comment]
Moved to bug report: https://github.com/fastlane/fastlane/issues/18069

**Comment by EmDee** [comment]
Extremely strange, but we are seeing this issue again 🤔 

What a weird coincidence that it's basically exactly one year ago that I've reported this problem... some magic happening in January or after NYE?
